src/Compilers/CSharp/Portable/Errors/MessageProvider.cs

```diff
@@ -57,7 +57,7 @@ public override string GetHelpLink(int code)
             return ErrorFacts.GetHelpLink((ErrorCode)code);
         }
 
-        public override string GetCategory(int code)
+        public override LocalizableString GetCategory(int code)
         {
             return ErrorFacts.GetCategory((ErrorCode)code);
```

Comment:
Did we change `ErrorFacts.GetCategory` to return a LocalizableString as well? Otherwise, we are just simulating a LocalizableString by wrapping a non-localizable string into a fixed value localizable string (always returns same value for all cultures).
We should instead change `ErrorFacts.GetCategory` to return a LocalizableString, which requires that `Diagnostic.CompilerDiagnosticCategory` should be a LocalizableString with a resource string in resx file. 

Additionally, currently `ErrorFacts.s_categoriesMap` is a placeholder map to store mappings from ErrorID to category strings. Its empty but likely will have key-value pairs once compiler diagnostics are categorized, instead of having a single category "Compiler". You should modify the comment in `ErrorFacts.CreateCategoriesMap` that the values of the map should represent name of localizable resource for category and probably change that method and the map on ErrorFacts to have LocalizableString values instead of string values.

All of the above also applies for VB compiler's `ErrorFactory` type and corresponding methods/field.